src/api/EscargotPublic.cpp

```diff
@@ -1146,6 +1146,62 @@ Evaluator::EvaluatorResult VMInstanceRef::executePendingJob()
 
 #ifdef ESCARGOT_DEBUGGER
 
+void DebuggerOperationsRef::BreakpointOperations::getStackTrace(DebuggerOperationsRef::DebuggerStackTraceDataVector& outStackTrace)
+{
+    ExecutionState* state = toImpl(m_executionState);
+    SandBox::StackTraceDataVector stackTraceData;
+
+    bool hasSavedStackTrace = SandBox::createStackTraceData(stackTraceData, *state, true);
+    ByteCodeLOCDataMap locMap;
+    size_t size = stackTraceData.size();
+
+    outStackTrace.clear();
+
+    for (uint32_t i = 0; i < size; i++) {
+        if ((size_t)stackTraceData[i].second.loc.actualCodeBlock != SIZE_MAX) {
+            ByteCodeBlock* byteCodeBlock = stackTraceData[i].second.loc.actualCodeBlock;
+            size_t line, column;
+
+            if ((size_t)stackTraceData[i].second.loc.index == SIZE_MAX) {
+                size_t byteCodePosition = stackTraceData[i].second.loc.byteCodePosition;
+
+                ByteCodeLOCData* locData;
+                auto iterMap = locMap.find(byteCodeBlock);
+                if (iterMap == locMap.end()) {
+                    locData = new ByteCodeLOCData();
+                    locMap.insert(std::make_pair(byteCodeBlock, locData));
+                } else {
+                    locData = iterMap->second;
+                }
+
+                ExtendedNodeLOC loc = byteCodeBlock->computeNodeLOCFromByteCode(state->context(), byteCodePosition, byteCodeBlock->m_codeBlock, locData);
+                line = (uint32_t)loc.line;
+                column = (uint32_t)loc.column;
+            } else {
+                line = (uint32_t)stackTraceData[i].second.loc.line;
+                column = (uint32_t)stackTraceData[i].second.loc.column;
+            }
+
+            outStackTrace.push_back(DebuggerStackTraceData(reinterpret_cast<WeakCodeRef*>(byteCodeBlock), line, column, stackTraceData[i].second.executionStateDepth));
+        }
+    }
+
+    for (auto iter = locMap.begin(); iter != locMap.end(); iter++) {
+        delete iter->second;
+    }
+
+    if (hasSavedStackTrace) {
+        Debugger* debugger = state->context()->debugger();
+        Debugger::SavedStackTraceData* savedStackTracePtr = debugger->activeSavedStackTrace()->begin();
+        Debugger::SavedStackTraceData* savedStackTraceEnd = debugger->activeSavedStackTrace()->end();
+
+        while (savedStackTracePtr < savedStackTraceEnd) {
+            outStackTrace.push_back(DebuggerStackTraceData(reinterpret_cast<WeakCodeRef*>(savedStackTracePtr->byteCodeBlock), savedStackTracePtr->line, savedStackTracePtr->column, SIZE_MAX));
+            savedStackTracePtr++;
+        }
```

Comment:
How about to fix this loop as follows?
(It's not mandatory, but looks more consistent compared to other code)
```c++
for (auto iter = debugger->activeSavedStackTrace()->begin();  iter != debugger->activeSavedStackTrace()->end(); iter++) {
}
```

Reply:
good idea. patch is updated.